packages/core/src/index.js

```diff
@@ -98,6 +98,10 @@ function replaceComponent(OldType, NewType, resetHookState) {
         vnode[VNODE_COMPONENT].constructor = NewType;
       }
 
+      vnode.type = NewType
+      vnode[VNODE_COMPONENT].__v = vnode;
+      vnode[VNODE_COMPONENT].__v.type = NewType;
```

Comment:
I think line 101 & 103 can be removed since they are redundant of each other and line 59.